xapian-letor/include/xapian-letor/feature.h

```diff
@@ -99,6 +100,16 @@ class XAPIAN_VISIBILITY_DEFAULT Feature {
     /// A bitmask of the statistics this Feature needs.
     stat_flags stats_needed;
 
+    /** A helper function for feature->get_value()
+     *
+     *  Checks if the term belongs to the title or is stemmed from the title.
+     */
+    inline bool
+    is_title_term(const std::string& term) const
+    {
+	return startswith(term, 'S') || startswith(term, "ZS");
+    }
+
```

Comment:
This really doesn't want need to go in the public API header.  It doesn't need the context of the class at all so just declare it as an inline (non-class) function in the file which uses it.